Updated docstring comments for exclude patterns and README pointer

michizhou opened the pull request:
Please fill in the fields below to submit a pull request.  The more information
that is provided, the better.

**Fixes issue #222**:

**Description of the changes being introduced by the pull request**:
Changed the description for exclude patterns to explain how gitignore-style patterns are compiled using PathSpec in the new implementation instead of Python fnmatch. Also updated the current pointer to the exclude patterns docstring in Available Settings section of the README.

**Please verify and check that the pull request fulfills the following
requirements**:

- [x] The code follows the [Code Style Guidelines](https://github.com/secure-systems-lab/code-style-guidelines#code-style-guidelines)
- [ ] Tests have been added for the bug fix or new feature
- [x] Docs have been added for the bug fix or new feature

michizhou committed:
Updated docstring comments for exclude patterns and README pointer

Changed description for exclude pattern to explain how gitignore-style patterns are compiled using PathSpec in the new implementation instead of Python fnmatch.Updated pointer to exclude patterns docstring in README (Available Settings).

lukpueh commented on README.md:
```diff
@@ -143,7 +143,7 @@ Also, setting values that contain colons are parsed as list.
 `ARTIFACT_EXCLUDE_PATTERNS` Specifies a list of glob patterns that can be used to
 exclude files from being recorded as materials or products. See [runlib
-docs for more details](https://github.com/in-toto/in-toto/blob/develop/in_toto/runlib.py#L93-L114).
+docs for more details](https://github.com/in-toto/in-toto/blob/develop/in_toto/runlib.py#L111-L129).
```
Why L111-L129?

michizhou replied:
I chose L111-L129 because those were the lines that referenced the info on exclude patterns in the docstring for the `record_artifacts_as_dict` function of `runlib.py`

lukpueh commented on in_toto/runlib.py:
```diff
@@ -101,11 +101,8 @@ def record_artifacts_as_dict(artifacts, exclude_patterns=None,
-            [seq]   matches any character in seq
-            [!seq]  matches any character not in seq
+      - Uses PathSpec to compile gitignore-style patterns, making use of the 
+        GitWildMatchPattern class (registered as 'gitwildmatch')
```
You changed the docs of an outdated `record_artifacts_as_dict` function, one that still used an `_apply_exclude_patterns` with `fnmatch`.

Please rebase you branch on top of [in-toto/in-toto@develop](https://github.com/in-toto/in-toto/tree/develop)! 

While even without rebasing a merge would actually pull your doc changes into the updated function, it is hard for me to review the docs if they don't correspond to the corresponding function.

michizhou replied:
I understand. I have rebased my branch accordingly, so you should be able to review my changes to the docs properly now.

lukpueh reviewed (changes requested):
+1 for the commit message (please also make sure that they don't exceed the suggested line length).
+1 for a descriptive branch name.

Please, address my comments below and I will re-review.

michizhou committed:
Updated README.md reference

Modified lines in pointer reference to docstring section on exclude patterns in
runlib file as an update for its associated link in the README

michizhou commented:
@lukpueh I just rebased this branch on top of [in-toto/in-toto@develop](https://github.com/in-toto/in-toto/tree/develop) and updated the README pointer to the exclude patterns docstring section in the `record_artifacts_as_dict` function of `runlib.py`. Please feel free to re-review my changes. Thanks!

lukpueh commented:
Thanks, @michizhou! Please also get rid of the merge commit in this PR. See https://github.com/in-toto/in-toto/pull/243#issuecomment-437824334.

michizhou commented:
@lukpueh I have just gotten rid of the merge commit as per your request. Please feel free to review my changes and let me know if any additional ones need to be made. Thanks.

lukpueh commented on README.md:
```diff
@@ -152,7 +152,7 @@ Also, setting values that contain colons are parsed as list.
 `ARTIFACT_EXCLUDE_PATTERNS` Specifies a list of glob patterns that can be used to
 exclude files from being recorded as materials or products. See [runlib
-docs for more details](https://github.com/in-toto/in-toto/blob/develop/in_toto/runlib.py#L93-L114).
+docs for more details](https://github.com/in-toto/in-toto/blob/develop/in_toto/runlib.py#L101-L119).
```
Since you're already at it, please replace the absolute link, which points to `runlib.py` in the `develop` branch, with a relative link:
`[runlib docs for more details](in_toto/runlib.py#L101-L119)`? 

Please also update the line numbers so that they match your changes below.

lukpueh reviewed (changes requested):
Thanks for addressing my comments. Please update the link in the README as suggested and we are good to merge. :)

michizhou committed:
Modified reference pointer in README

Replaced absolute link in README reference pointer with a relative link so that
pointer no longer has to point to reference file in specific branch

michizhou commented:
@lukpueh I have edited the README to reflect your suggested changes. Please let me know if there is anything else I may need to add or change. Thanks!

lukpueh commented:
Thanks, @michizhou!
